File: front/src/types/meeting.ts
export interface Participant {
  id: string
  name: string
  email: string
  created_at?: string
}

export interface Meeting {
  id: string
  title: string
  description?: string | null
  link_to_call?: string | null
  place?: string | null
  start_time?: string | null
  end_time?: string | null
  created_at: string
  updated_at: string
  participants: Participant[]
}

export interface CreateMeetingPayload {
  title: string
  description?: string
  link_to_call?: string
  place?: string
  start_time?: string
  end_time?: string
  participant_ids?: string[]
  new_participants?: { name: string; email: string }[]
}

export interface CreateParticipantPayload {
  name: string
  email: string
}

File: back/app/schemas/meeting.py
import uuid
from datetime import datetime
from typing import List, Optional

from app.schemas.participant import ParticipantCreate, ParticipantResponse
from pydantic import BaseModel, ConfigDict, Field


class MeetingBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    link_to_call: Optional[str] = Field(None, max_length=2048)
    place: Optional[str] = Field(None, max_length=255)
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None


class MeetingCreate(MeetingBase):
    participant_ids: List[uuid.UUID] = Field(default_factory=list)
    new_participants: List[ParticipantCreate] = Field(default_factory=list)


class MeetingResponse(MeetingBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    participants: List[ParticipantResponse] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)
